### PR TITLE
CPU Plugin refactoring: Transition from Intel MKL-DNN to oneDNN

### PR DESCRIPTION
**Pull request by vladislav-volkov**
### Details:
 - CPU Plugin refactoring: Transition from Intel MKL-DNN to oneDNN

### Tickets:
 - CVS-74811

### CODE REVIEW

File: docs/glossary.md [hunk 1]
```diff
@@ -27,6 +27,7 @@
 | LRN               | Local Response Normalization |
 | mAP               | Mean Average Precision |
 | Intel(R) MKL-DNN  | Intel(R) Math Kernel Library Deep Neural Networks |
+| Intel(R) OneDNN   | Intel(R) OneAPI Deep Neural Network Library |
```
Comment (nsemaev) [inline docs/glossary.md:30]: Perhaps the MKL-DNN description should be removed?

File: tools/legacy/benchmark_app/main.cpp [hunk 2]
```diff
@@ -171,11 +171,11 @@ int main(int argc, char* argv[]) {
 
         Core ie;
         if (FLAGS_d.find("CPU") != std::string::npos && !FLAGS_l.empty()) {
-            // CPU (MKLDNN) extensions is loaded as a shared library and passed as a
+            // CPU plugin extensions is loaded as a shared library and passed as a
             // pointer to base extension
             const auto extension_ptr = std::make_shared<InferenceEngine::Extension>(FLAGS_l);
             ie.AddExtension(extension_ptr);
-            slog::info << "CPU (MKLDNN) extensions is loaded " << FLAGS_l << slog::endl;
+            slog::info << "CPU plugin extensions is loaded " << FLAGS_l << slog::endl;
```
Comment (nsemaev) [inline tools/legacy/benchmark_app/main.cpp:178]: Misspelling "**are** loaded"

**Review by nsemaev — COMMENTED** [review]
IE Tests reviewed

File: docs/glossary.md (hunk 1 above)
Comment (vladislav-volkov) [reply docs/glossary.md:30]: removed

File: tools/legacy/benchmark_app/main.cpp (hunk 2 above)
Comment (vladislav-volkov) [reply tools/legacy/benchmark_app/main.cpp:178]: fixed

File: src/plugins/intel_cpu/src/nodes/def_conv.cpp [hunk 3]
```diff
@@ -6,18 +6,18 @@
...
 #include "memory_desc/dnnl_blocked_memory_desc.h"
 
-using namespace mkldnn;
+using namespace dnnl;
 using namespace InferenceEngine;
-using namespace mkldnn;
-using namespace mkldnn::impl;
-using namespace mkldnn::impl::cpu::x64;
-using namespace mkldnn::impl::utils;
+using namespace dnnl;
```
[the remark this answers is not in the document]
Comment (vladislav-volkov) [reply src/plugins/intel_cpu/src/nodes/def_conv.cpp:17]: removed

**Review by maxnick — APPROVED** [review]

**Review by dmitry-gorokhov — APPROVED** [review]

File: src/frontends/onnx/frontend/src/op/org.openvinotoolkit/detection_output.cpp [hunk 4]
```diff
@@ -25,7 +25,7 @@ OutputVector detection_output(const Node& node) {
     attrs.top_k = node.get_attribute_value<int64_t>("top_k", -1);
     attrs.variance_encoded_in_target = node.get_attribute_value<int64_t>("variance_encoded_in_target", 0);
     // spec says keep_top_k is an array of ints, but some models use a single int
-    // also mkldnn expects single integer
+    // also onednn expects single integer
```
Comment (dmitry-gorokhov) [inline src/frontends/onnx/frontend/src/op/org.openvinotoolkit/detection_output.cpp:28]: onednn -> CPU plugin

File: src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp [hunk 5]
```diff
@@ -318,7 +318,7 @@ TEST_F(myriadLayersTestsProposal_smoke, CaffeNoClipBeforeNms) {
         0, -107.360f, 52.610f,  326.355f, 222.650f,
         0, -223.833f, 21.986f,  428.385f, 299.044f,
 
-        // swapped on mkldnn
+        // swapped on onednn
```
Comment (dmitry-gorokhov) [inline src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp:321]: onednn -> CPU plugin

File: src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp [hunk 6]
```diff
@@ -29,7 +29,7 @@ class myriadLayersTestsProposal_smoke : public myriadLayersTests_nightly {
 
 
  /**
-  * to generate reference mkldnn-can be used, but even in hetero cpu,cpu it cannot handle fp16 input
+  * to generate reference onednn-can be used, but even in hetero cpu,cpu it cannot handle fp16 input
```
Comment (dmitry-gorokhov) [inline src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp:32]: onednn -> CPU plugin

**Comment by dmitry-gorokhov** [comment]
@vladislav-volkov please also rebase on latest master

File: src/frontends/onnx/frontend/src/op/org.openvinotoolkit/detection_output.cpp (hunk 4 above)
Comment (vladislav-volkov) [reply src/frontends/onnx/frontend/src/op/org.openvinotoolkit/detection_output.cpp:28]: fixed

File: src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp (hunk 5 above)
Comment (vladislav-volkov) [reply src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp:321]: fixed

File: src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp (hunk 6 above)
Comment (vladislav-volkov) [reply src/tests_deprecated/functional/vpu/common/layers/myriad_layers_proposal_test.cpp:32]: fixed

**Review by nsemaev — APPROVED** [review]

**Pull request merged**